src/lambda_streams/Animals.java

```diff
@@ -1,62 +1,173 @@
+        if(mutate)
+        {
+            animals = animaList.stream()
+                    .map(x -> x.substring(0, 1).toUpperCase() + x.substring(1).toLowerCase())
+                    .collect(Collectors.toList());
+        } else
+            {
+            List<String> animaList2 = animaList;
+            return animaList2.stream()
+                    .map(x -> x.substring(0, 1).toUpperCase() + x.substring(1).toLowerCase())
+                    .collect(Collectors.toList());
+            }
         return animals;
     }
 
-    static String addAnimal(String animal) {
+    static String addAnimal(String animal)
+    {
         //add an animal to the animal list.
+        ArrayList<String> animaList3 = new ArrayList<>(animals);
+//        ArrayList<String> updatedAnimals = new ArrayList<String>(animals);
+        animaList3.add(animal);
+        animals = animaList3;
         return animal;
     };
 
 
-    static List<String> lowerFirst(List<String> animaList, boolean mutate) {
+    static List<String> lowerFirst(List<String> animaList, boolean mutate)
+    {
         //lowercase the first letter, and uppercase the rest of the letters, using streams.  Also, depending on the value of the boolean flag 'mutate', mutate the original animals list, or perform your stream operations on a 'copy' of the animals list.  return the list out of hte function in both cases.
+        if(mutate)
+        {
+            animals = animaList.stream()
+                    .map(x -> x.substring(0, 1).toLowerCase() + x.substring(1).toUpperCase())
+                    .collect(Collectors.toList());
+        } else {
+            List<String> animaList3 = animaList;
+            return animaList3.stream()
+                    .map(x -> x.substring(0, 1).toLowerCase() + x.substring(1).toUpperCase())
+                    .collect(Collectors.toList());
+        }
         return animaList;
     }
 
-    static List<String> flipAnimals(boolean mutate) {
+    static List<String> flipAnimals(boolean mutate)
+    {
         //reverse the order of the animals in the animal list.  If the booleaen parameter is true, reverse the static animals array list by mutating the array.  if the mutate boolean is false, flip a 'copy' of the animals list, then return that list of flipped animals, WITHOUT mutating the static animals array. return the flipped list in both cases.
 
-        List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals","only","if","flag","true");
-        return instructions;
+//        List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals","only","if","flag","true");
+//        return instructions;
+        if(mutate)
+        {
+            Collections.reverse(animals);
+        } else {
+            List<String> animaList4 = animals;
+            Collections.reverse(animaList4);
```

Comment:
Because 'animalList4' is being set with a reference to 'animals', this line flips the values of 'animals' even though 'mutate' was set to "false".  How might this be refactored to avoid this undesirable behavior?